### PR TITLE
Fix bugs in the dynamic validators

### PR DESCRIPTION
**Pull request by sgkim126**
It depends on https://github.com/CodeChain-io/codechain/pull/1638

### CODE REVIEW

**Comment by sgkim126** [comment]
@majecty Would you check whether https://github.com/CodeChain-io/codechain/pull/1639/commits/d8eaa23933802375fc09487de4a70d492607783d it's correct? I think it should use the hash of the grandparent.

**Comment by sgkim126** [comment]
@majecty @foriequal0 https://github.com/CodeChain-io/codechain/pull/1639/commits/fe7fbc8efc0f5fd916bc0ce57824173dc62e6949 changed the spec of the dynamic validators

File: core/src/consensus/stake/action_data.rs [hunk 1]
```diff
@@ -327,35 +366,42 @@ impl Validators {
             // neglecting validators
             *weight = weight.saturating_sub(min_delegation * 2);
         }
-        self.0.sort();
+        if self.0.iter().any(|validator| validator.weight != 0) {
+            self.0.sort_unstable();
+        } else {
+            self.0.iter_mut().for_each(Validator::reset);
```
Comment (majecty) [inline core/src/consensus/stake/action_data.rs]: Should we sort again after the reset?

Comment (sgkim126) [reply core/src/consensus/stake/action_data.rs]: You're right. I'll fix it.

**Comment by sgkim126** [comment]
@foriequal0 I'll merge it first please make a PR if you find bugs.

**Comment by foriequal0** [comment]
Okay. LGTM.

**Pull request merged**